enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,255 @@
+- When the newly spawned container is failing to upload the data then it can be running for some longer time (couple of hours). 
+
+
+## Design Details
+
+### Open Questions
+- How to solve the problem with only one Insights analysis result available in a cluster? How to avoid the "uncertainty" of the Insights analysis results described in the [Risk and Mitigations](#risks-and-mitigations) section? 
+- Should the newly created gathering process update the clusteroperator conditions (e.g it fails to upload the data, should it update the `Degraded` status as it is today)? This can probably be racy too or it can lead to temporary and inconsistent conditions state. 
+- What all is required to spawn an extra container? Do we need a new image and a new binary (note that the Insights operator currently includes some additional features - such as pulling SCA certificates - that are not required for a new container/process)?
+- Is some clean-up of the old `datagather.insights.openshift.io` CRs (and completed containers/pods) required? I think it would make sense to keep them only for 24h.
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API behind the `TechPreviewNoUpgrade` feature gate.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+The `TechPreviewNoUpgrade` feature gate requirement is removed. 
+
+Other than that we would like to:
+- get some time to gather feedback and some user experience
+- ensure stability
+- do more e2e testing
+
+#### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+No special upgrade or downgrade strategy should be needed. The new `datagather.insights.openshift.io` CRD will be introduced with an upgrade to given version (allowing users to run on demand Insights data gathering). This CRD will be removed during the downgrade. 
+
+### Version Skew Strategy
+
+### Operational Aspects of API Extensions
+
+This API is provided exclusively for the Insights operator and there will be no other component depending on this new API.
+
+#### Failure Modes
+
+TBD
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+As noted in the [Graduation criteria](#graduation-criteria) section, the first API version will be behind the `TechPreviewNoUpgrade` feature gate.
+
+## Alternatives
+
+- Current alternative is to delete the Insights operator Pod and let the replication controller start a new Pod that will run new data gathering almost immediately. 
+- Run the on-demand gathering within an existing container. This approach would use very similar `datagather.insights.openshift.io` CR (as mentioned above) to run the on-demand Insights data gathering, but the operator wouldn't spawn any new container. 
```

Comment:
I think I would still prefer this alternative, because it seems more straightforward and removes the major questions/issues described in the `Risks and Mitigations` section.

Reply:
Then you have a blocking job within your operator, creating new pods has benefits as well as the drawbacks, it provides way more observability into what's happening for a single DataGather vs it running in the process within the operator

For example, I can see my `DataGather` maps to a particular pod and therefore can observe it's logs to debug, rather than trying to detangle from the operator logs

Reply:
It wouldn't be blocking. It would include interrupting of any active gathering process, but yeah otherwise I agree with you (with the observability aspect).

Reply:
I believe we can resolve this discussion.